.github/workflows/primary.yml

```diff
@@ -0,0 +1,59 @@
+name: Pre-Release
+
+on:
+  pull_request:
+    branches:
+      - primary
+jobs:
+  Primary-Merge-Checks:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python 3.8
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+      - name: Install and set up Poetry
+        run: |
+          curl -fsS -o get-poetry.py https://raw.githubusercontent.com/sdispater/poetry/master/get-poetry.py
+          python get-poetry.py --preview -y
+          source $HOME/.poetry/env
+      - name: Set up cache
+        uses: actions/cache@v1
+        with:
+          path: .venv
+          key: venv-${{ runner.os }}-${{ steps.full-python-version.outputs.version }}-${{ hashFiles('**/poetry.lock') }}
+      - name: Upgrade pip
+        run: |
+          source $HOME/.poetry/env
+          poetry run python -m pip install pip -U
+      - name: Install dependencies
```

Comment:
do we need to install dependencies if only checking version?

Reply:
yes because we import the snappiershot module to get the current version. This is done in line 45